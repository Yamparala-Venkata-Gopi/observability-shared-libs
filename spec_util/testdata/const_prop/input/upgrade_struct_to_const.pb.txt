# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    struct_template: {
      # Both fields refer to constant -- the whole struct should be upgraded to
      # a constant.
      field_templates: {
        key: "field1"
        value: {
          ref: {
            method_index: 0
            arg_ref: {
              key: "primitive_constant_arg"
              data_ref: {
                primitive_ref: {}
              }
            }
          }
        }
      }
      field_templates: {
        key: "field2"
        value: {
          ref: {
            method_index: 0
            arg_ref: {
              key: "struct_constant_arg"
              data_ref: {
                struct_ref: {
                  full_struct: {}
                }
              }
            }
          }
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    struct_template: {
      # All fields are constant already -- the whole struct should be upgraded
      # to a constant.
      field_templates: {
        key: "field1"
        value: {
          value: {
            primitive: {
              int64_value: {
                value: 522
              }
            }
          }
        }
      }
      field_templates: {
        key: "field2"
        value: {
          value: {
            primitive: {
              int32_value: {
                value: 1203
              }
            }
          }
        }
      }
    }
  }
}
