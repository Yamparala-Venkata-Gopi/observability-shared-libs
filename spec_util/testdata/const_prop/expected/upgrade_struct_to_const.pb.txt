# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    value: {
      struct: {
        fields: {
          key: "field1"
          value: {
            primitive: {
              type_hint: "string_constant_1"
              string_value: {
                value: "7c73565e-0d62-49b0-84e7-3063f4d53d9c"
              }
            }
          }
        }
        fields: {
          key: "field2"
          value: {
            struct: {
              fields: {
                key: "foo"
                value: {
                  primitive: {
                    type_hint: "string_constant_2"
                    string_value: {
                      value: "1aa4e597-a29d-42d4-8134-3e4ee73aa7a1"
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    value: {
      struct: {
        fields: {
          key: "field1"
          value: {
            primitive: {
              int64_value: {
                value: 522
              }
            }
          }
        }
        fields: {
          key: "field2"
          value: {
            primitive: {
              int32_value: {
                value: 1203
              }
            }
          }
        }
      }
    }
  }
}
